lms/static/scripts/frontend_apps/utils/fetch.ts

```diff
@@ -30,6 +33,11 @@ export type FetchResult<T> = {
  */
 export type Fetcher<T> = (signal: AbortSignal) => Promise<T>;
 
+export type UseFetchOptions = {
+  /** Whether this fetch should be reported to the InitialLoadingContext */
+  reportToInitialLoading?: boolean;
```

Comment:
This assumes that we'd only ever want to report render-blocking loads to the context. I wonder if the flag should be named `blockInitialDisplay` or `blockInitialRender` or something.